src/components/application_manager/src/commands/hmi/navi_set_video_config_request.cc

```diff
@@ -0,0 +1,150 @@
+    return;
+  }
+
+  subscribe_on_event(hmi_apis::FunctionID::Navigation_SetVideoConfig,
+                     correlation_id());
+  SendRequest();
+}
+
+void NaviSetVideoConfigRequest::on_event(const event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ApplicationSharedPtr app =
+      application_manager_.application_by_hmi_app(application_id());
+  if (!app) {
+    LOG4CXX_ERROR(logger_,
+                  "Application is not found, abort NaviSetVideoConfigRequest");
+    return;
+  }
+
+  const smart_objects::SmartObject& message = event.smart_object();
+  switch (event.id()) {
+    case hmi_apis::FunctionID::Navigation_SetVideoConfig: {
+      const hmi_apis::Common_Result::eType code =
+          static_cast<hmi_apis::Common_Result::eType>(
+              message[strings::params][hmi_response::code].asInt());
+      bool result = false;
+      std::vector<std::string> rejected_params;
+
+      if (code == hmi_apis::Common_Result::SUCCESS) {
+        LOG4CXX_DEBUG(logger_, "Received SetVideoConfig success response");
+        result = true;
+      } else {
+        LOG4CXX_DEBUG(logger_,
+                      "Received SetVideoConfig failure response (" << event.id()
+                                                                   << ")");
+        result = false;
+        if (message[strings::msg_params].keyExists(strings::rejected_params)) {
+          const smart_objects::SmartArray* list =
+              message[strings::msg_params][strings::rejected_params].asArray();
+          if (list != NULL) {
+            for (unsigned int i = 0; i < list->size(); i++) {
+              const std::string& param = (*list)[i].asString();
+              // Make sure that we actually sent the parameter in the request
+              if ((*message_)[strings::msg_params].keyExists(strings::config) &&
+                  (*message_)[strings::msg_params][strings::config].keyExists(
+                      param)) {
+                rejected_params.push_back(param);
+              }
+            }
+          }
+        }
+      }
+      application_manager_.OnStreamingConfigured(
+          application_id(),
+          protocol_handler::ServiceType::kMobileNav,
+          result,
+          rejected_params);
+      break;
+    }
+    default:
+      LOG4CXX_ERROR(logger_, "Received unknown event" << event.id());
+      break;
+  }
+}
+
+void NaviSetVideoConfigRequest::onTimeOut() {
+  LOG4CXX_WARN(logger_, "Timed out while waiting for SetVideoConfig response");
+
+  ApplicationSharedPtr app =
+      application_manager_.application_by_hmi_app(application_id());
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "Application is not found");
+    return;
+  }
+
+  std::vector<std::string> empty;
+  application_manager_.OnStreamingConfigured(
+      application_id(),
+      protocol_handler::ServiceType::kMobileNav,
+      false,
+      empty);
+
+  application_manager_.TerminateRequest(
```

Comment:
Response to mobile proxy (which is actually Start Service NACK) is triggered by OnStreamingConfigured().